controllers/ciresource_fsm.go

```diff
@@ -172,8 +172,14 @@ func (f *CIResourceFSM) handleStateInUse(context CIResourceFSMContext) (time.Dur
 		if context.CIPool.IsFallbackPool() && context.CIResource.Status.Address == "" && context.CIResource.Status.ResourceId == fallbackResourceID {
 			return f.TriggerEvent("fallback-provisioning")
 		}
+
 		// CIR's can only be held "inuse" for a limited amount of time
-		if time.Now().Sub(context.CIResource.Status.LastUpdated.Time) > context.CIPool.Spec.Timeout.Duration {
+		currentTime := time.Now()
+		lastUpdate := context.CIResource.Status.LastUpdated.Time
+		elapsed := currentTime.Sub(lastUpdate)
+		f.logger.Info("resource timeout check", "current", currentTime.String(), "lastUpdate", lastUpdate.String(), "elapsed (sec)", elapsed.Seconds(), "poolTimeout", context.CIPool.Spec.Timeout.Duration)
```

Comment:
This could potentially add a lot to the logging, I'm wondering if we would be better to move it to replace the log message inside the of conditional?